pkg/monitor/cluster/debugpods_test.go

```diff
@@ -0,0 +1,82 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	ktesting "k8s.io/client-go/testing"
+
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+)
+
+func TestEmitDebugPodsCount(t *testing.T) {
+	ctx := context.Background()
+
+	cli := fake.NewSimpleClientset(
+		&corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "master-1",
+			},
+		},
+		&corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "master-2",
+			},
+		},
+	)
+
+	now := time.Now()
+	longAgo := time.Date(1991, time.August, 24, 0, 0, 0, 0, time.UTC)
+	cli.PrependReactor("list", "events", func(action ktesting.Action) (handled bool, ret kruntime.Object, err error) {
+		return true, &corev1.EventList{Items: []corev1.Event{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "master-1-started",
+					Namespace: "default",
+				},
+				Reason: "Started",
+				InvolvedObject: corev1.ObjectReference{
+					Kind: "Pod",
+					Name: "master-1-debug",
+				},
+				LastTimestamp: metav1.NewTime(longAgo),
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "master-2-started",
+					Namespace: "default",
+				},
+				Reason: "Started",
+				InvolvedObject: corev1.ObjectReference{
+					Kind: "Pod",
+					Name: "master-2-debug",
+				},
+				LastTimestamp: metav1.NewTime(now),
+			},
+		}}, nil
+	})
```

Comment:
Anonymous functions can become really ugly in the long run. I propose a simple change to extract it to a variable.
```go
prependReactorFn := func(action ktesting.Action) (handled bool, ret kruntime.Object, err error) {
        return true, &corev1.EventList{Items: []corev1.Event{
	        {
		        ObjectMeta: metav1.ObjectMeta{
			        Name:      "master-1-started",
			        Namespace: "default",
		        },
		        Reason: "Started",
		        InvolvedObject: corev1.ObjectReference{
			        Kind: "Pod",
			        Name: "master-1-debug",
		        },
		        LastTimestamp: metav1.NewTime(longAgo),
	        },
	        {
		        ObjectMeta: metav1.ObjectMeta{
			        Name:      "master-2-started",
			        Namespace: "default",
		        },
		        Reason: "Started",
		        InvolvedObject: corev1.ObjectReference{
			        Kind: "Pod",
			        Name: "master-2-debug",
		        },
		        LastTimestamp: metav1.NewTime(now),
	        },
        }}, nil
}
cli.PrependReactor("list", "events", prependReactorFn)
```

Reply:
agreed 😎